addon/utils/array-helpers.js

```diff
@@ -167,3 +167,29 @@ export function uniqueValues(arr) {
 
   return [...new Set(arr)];
 }
+
+export function findBy(arr, key, searchValue) {
+  if (arr !== null && typeof arr === 'object') {
+    if ('slice' in arr) {
+      arr = arr.slice();
+    }
+  }
+
+  if (!Array.isArray(arr)) {
```

Comment:
this does not track for me at all.
if the given input is not an array, or cannot be extracted into an array, then _it itself_ is returned. 
that seems like a bug to me.
consider this test passing

```js
  test('findBy when input is not array and cannot be converted into an array', function (assert) {
    const arr = {};
    const result = findBy(arr, 'id', 3); // doesn't matter what filters you apply here, you always get the input "array" back.
    assert.strictEqual(result, arr);
  });
```

i think an Error should be raised here instead.

Reply:
whatever we end up doing - return a blank array in case of bad input, raise an error, or whatever else - the function should not return itself if it's not an array/array-like.

perhaps returning an empty array and adding a type-checking [assertion](https://developer.mozilla.org/en-US/docs/Web/API/console/assert) here might be the least invasive solution.